cpp/src/groupby/sort/group_collect.cu

```diff
@@ -17,30 +17,98 @@
 #include <cudf/column/column_factories.hpp>
 #include <cudf/column/column_view.hpp>
 #include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/detail/copy_if.cuh>
 #include <cudf/detail/gather.cuh>
+#include <cudf/strings/detail/utilities.cuh>
 #include <cudf/types.hpp>
 #include <cudf/utilities/span.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
 
+#include <memory>
+
 namespace cudf {
 namespace groupby {
 namespace detail {
+/**
+ * @brief Purge null entries in grouped values, and adjust group offsets.
+ */
+std::pair<std::unique_ptr<column>, std::unique_ptr<column>> purge_null_entries(
+  column_view const &values,
+  column_view const &offsets,
+  size_type num_groups,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource *mr)
+{
+  auto values_device_view = column_device_view::create(values, stream);
+
+  auto not_null_pred = [d_value = *values_device_view] __device__(auto i) {
+    return d_value.is_valid_nocheck(i);
+  };
+
+  // Purge null entries in grouped values.
+  auto null_purged_entries =
+    cudf::detail::copy_if(table_view{{values}}, not_null_pred, stream, mr)->release();
+
+  std::unique_ptr<column> &null_purged_values = null_purged_entries[0];
+
+  // Recalculate offsets after null entries are purged.
+  auto null_purged_sizes = make_numeric_column(
+    data_type{type_to_id<size_type>()}, num_groups, mask_state::UNALLOCATED, stream, mr);
```

Comment:
This appears to be a temporary data column only needed to build the `null_purged_offsets` output. Therefore, it should use `rmm::mr::get_current_device_resource()` instead of the passed in `mr`. It may be better to just use a `device_uvector` here instead since there is no need to make a column.

Reply:
Yes, thanks for suggestion. I've replaced the temporary column with `rmm::device_uvector`.